Add native VOL checks to deprecated functions

mattjala opened the pull request:
Fix for #3620

Currently causes the HDF EOS tests to fail in `HE5_sw_writedata`, upon tyring to reclaim data from a variable length datatype during `HE5_PRwrite`.

mattjala committed:
Add native VOL checks to deprecated functions

mattjala committed:
Remove unneeded native VOL checks

mattjala commented:
> Currently causes the HDF EOS tests to fail in `HE5_sw_writedata`, upon tyring to reclaim data from a variable length datatype during `HE5_PRwrite`.

Fixed by removing the native VOL check in `H5Dvlen_reclaim`, which turned out to be unnecessary anyway.

fortnern commented on src/H5Odeprec.c:
```diff
@@ -127,6 +129,19 @@ H5O__iterate1_adapter(hid_t obj_id, const char *name, const H5O_info2_t *oinfo2,
+        HGOTO_ERROR(H5E_ARGS, H5E_BADTYPE, FAIL, "invalid location identifier");
+
+    /* Check if using native VOL connector */
+    if (H5VL_object_is_native(vol_obj, &is_native) < 0)
```
This is probably fine, but I think it would be better to put this check at the top level, in HOvisit1(), etc. Even though it needs to be in more places I think it's better to catch it before it goes into the VOL connector at all and give a cleaner error message.

mattjala committed:
Move native checks to top level calls

mattjala commented:
I moved the native VOL checks from the internal `H5O__iterate1_adapter`/`H5O__get_info_old` functions up into the API functions that use them internally. Also renamed the boolean `is_native -> is_native_vol_obj` for consistency within the file.
